### PR TITLE
Updated code to use HashMap instead of ImmutableMap.Builder

### PR DESCRIPTION
**Pull request by sagarkapare**
In distributed mode the RunId is passed to the programs from AbstractProgramTwillRunnable. So the RunId key is always present in the options. If we use ImmutableMap.Builder to recreate the options in the ProgramRunner, in distributed mode it fails with "duplicate key" exception. There is a bigger refactoring needs to be done around how RunId should be propagated as a part of jira https://issues.cask.co/browse/CDAP-2039.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/AbstractProgramWorkflowRunner.java [hunk 1]
```diff
@@ -32,10 +32,14 @@
 import co.cask.cdap.internal.app.runtime.SimpleProgramOptions;
 import com.google.common.base.Throwables;
 import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
 import com.google.common.util.concurrent.Futures;
 import com.google.common.util.concurrent.SettableFuture;
+import jline.internal.Log;
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/AbstractProgramWorkflowRunner.java]: Remove all log related imports.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/AbstractProgramWorkflowRunner.java [hunk 2]
```diff
@@ -54,6 +58,7 @@
  * The {@link RuntimeContext} is blocked until completion of the associated program.
  */
 public abstract class AbstractProgramWorkflowRunner implements ProgramWorkflowRunner {
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractProgramWorkflowRunner.class);
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/AbstractProgramWorkflowRunner.java]: unused?

**Comment by bdmogal** [comment]
Minor comments, then LGTM

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/AbstractProgramWorkflowRunner.java (hunk 1 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/AbstractProgramWorkflowRunner.java]: Fixed.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/AbstractProgramWorkflowRunner.java (hunk 2 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/AbstractProgramWorkflowRunner.java]: My bad. Removed.

**Comment by sagarkapare** [comment]
Build is running here - http://builds.cask.co/browse/CDAP-DUT1381-2

**Comment by sagarkapare** [comment]
Build passed. Merging.

**Pull request merged**